src/test/compile-fail/unreachable-arm.rs

```diff
@@ -12,7 +12,16 @@
 
 #![feature(box_patterns)]
 #![feature(box_syntax)]
+#![allow(dead_code)]
+#![deny(unreachable_patterns)]
 
-enum foo { a(Box<foo>, isize), b(usize), }
+enum Foo { A(Box<Foo>, isize), B(usize), }
+
+fn main() {
+    match Foo::B(1) {
+        Foo::B(_) | Foo::A(box _, 1) => { }
+        Foo::A(_, 1) => { }
+        _ => { }
```

Comment:
Why is this test written with "error-pattern" and not `//~ ERROR`? Can we convert it?